lib/matplotlib/axes/_axes.py

```diff
@@ -3332,14 +3331,14 @@ def errorbar(self, x, y, yerr=None, xerr=None,
             if key in kwargs:
                 eb_lines_style[key] = kwargs[key]
 
-        # set up cap style dictionary
+        # Make the style dict for the caps, ejecting any marker information
+        # from format string.
         eb_cap_style = dict(base_style)
-        # eject any marker information from format string
         eb_cap_style.pop('marker', None)
-        eb_lines_style.pop('markerfacecolor', None)
-        eb_lines_style.pop('markeredgewidth', None)
-        eb_lines_style.pop('markeredgecolor', None)
-        eb_cap_style.pop('ls', None)
+        eb_cap_style.pop('markersize', None)
+        eb_cap_style.pop('markerfacecolor', None)
+        eb_cap_style.pop('markeredgewidth', None)
+        eb_cap_style.pop('markeredgecolor', None)
```

Comment:
Not quite following why all this popping needs to be done twice...

Reply:
What do you mean?

Reply:
Ah, I see, `base_style` is no longer used to create the line, so we pop everything before creating the derivative styles.